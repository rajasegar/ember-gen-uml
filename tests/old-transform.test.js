/* globals test expect */
const transform = require("../src/transform");

  test("should generate transform public property properly", function() {

    const input = `
export default Component.extend({
  prop1: true
})`;

    expect(transform(input)).toMatchSnapshot();
  });

  test("should generate transform multiple properties properly", function() {

    const input = `
export default Component.extend({
  prop1: true,
  prop2: 'hello'
})`;

    expect(transform(input)).toMatchSnapshot();
  });


  test("should generate transform private property properly", function() {

    const input = `
export default Component.extend({
  _prop1: true
})`;


    expect(transform(input)).toMatchSnapshot();

  });


  test("should generate transform public method properly", function() {

    const input = `
export default Component.extend({
  method1() {
  }
})`;


    expect(transform(input)).toMatchSnapshot();

  });

test("should generate transform private method properly", function() {

    const input = `
export default Component.extend({
  _method1() {
  }
})`;

    expect(transform(input)).toMatchSnapshot();
  });


test("should generate transform service properties  properly", function() {

    const input = `
export default Component.extend({
  marketplace: service(),
  overlay: service()
})`;

    expect(transform(input)).toMatchSnapshot();
  });


test("should generate transform extends properly", function() {

    const input = `
export default Component.extend(Component1, Mixin1, {
  marketplace: service(),
  overlay: service()
})`;

    expect(transform(input)).toMatchSnapshot();
  });

