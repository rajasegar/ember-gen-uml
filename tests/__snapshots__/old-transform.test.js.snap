// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate transform extends properly 1`] = `
"
@startuml
Component1 <|-- Component
Mixin1 <|-- Component
class marketplace << (S, #FF7700) >>
Component ..> marketplace : service
class overlay << (S, #FF7700) >>
Component ..> overlay : service
class Component {
  +marketplace
  +overlay
}

@enduml"
`;

exports[`should generate transform multiple properties properly 1`] = `
"
@startuml


class Component {
  +prop1
  +prop2
}

@enduml"
`;

exports[`should generate transform private method properly 1`] = `
"
@startuml


class Component {
  -_method1()
}

@enduml"
`;

exports[`should generate transform private property properly 1`] = `
"
@startuml


class Component {
  -_prop1
}

@enduml"
`;

exports[`should generate transform public method properly 1`] = `
"
@startuml


class Component {
  +method1()
}

@enduml"
`;

exports[`should generate transform public property properly 1`] = `
"
@startuml


class Component {
  +prop1
}

@enduml"
`;

exports[`should generate transform service properties  properly 1`] = `
"
@startuml

class marketplace << (S, #FF7700) >>
Component ..> marketplace : service
class overlay << (S, #FF7700) >>
Component ..> overlay : service
class Component {
  +marketplace
  +overlay
}

@enduml"
`;
